paddle/operators/fill_op.h

```diff
@@ -0,0 +1,42 @@
+/* Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserve.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License. */
+
+#pragma once
+#include <vector>
+
+#include "paddle/framework/op_registry.h"
+
+namespace paddle {
+namespace operators {
+template <typename T>
+class FillOpKernelBase : public framework::OpKernel {
```

Comment:
Maybe the base class FillOpKernelBase is a little complex, just implementing data fill in  FillOpGPUKernel and FillOpCPUKernel will be fine.